### DISCUSSION TITLE
Is there a way to prevent auto-erasure of the sketches on close?

### DISCUSSION THREAD

**Discussion by nickludlam** [pubpub-zz/ppInk#247] [Q&A]
Hi, my preference in how to use this would be to treat it as a persistent whiteboard where I'm sketching ideas over time, and I wanted to see if there's a way to prevent it from auto-erasing the current sketch on close? I realise there's an ability to load the previously auto-saved sketch, but is there any way to make this automatic? Or would this need a checkbox added to the preferences?

**Comment by nickludlam** [comment]
There is a bool called `EraseOnLoosingFocus` from https://github.com/pubpub-zz/ppInk/blob/ee37adbcf0045706ba2a6d108e03d085e10d15a2/src/Root.cs#L439 and it's controlled via the config file, but it doesn't seem to be referenced anywhere else in the code.

**Comment by pubpub-zz** [comment]
The EraseOnLoosingFocus as it is named is affecting behavior when ppInk looses focus. I don't think it is what you are looking for.
When toolbar is closed the program is in idle and all image is closed. If you wan to stop inking but keep the drawings displayed, you just need to engage Pointer mode (clicking on the the cursor like icon)
If you have by error closed the toolbar, when reopening it, you can just reload your previous strokes using the load button
Finally if you want to use it as a real whiteboard, you may be interrested to use ppink in window mode.

Do not hesitate to star the project 😊

**Reply by nickludlam** [reply]
This is almost what I need, but I use whiteboard software almost like taking notes. I want to be able to add or change a diagram I sketched 2 minutes ago, and not need to draw it all again. How difficult would it be to stop the images being closed/deallocated when you press escape and return to using other apps?

**Reply by nickludlam** [reply]
I've made the changes to my own fork here:  https://github.com/nickludlam/ppInk/commit/4af54bf0adb09939b488f7e7a2ead858d32aba65

**Reply by pubpub-zz** [reply]
I'm very happy you've been able to modify the software : I can assume my code is not too obvious🤞😊
for my understanding, why are you not just loading the autosave.strokes.txt if PersistenceDrawing==True ?
Else can you propose it as a PR in order to not loose your work : I'm ok to include it 👍

**Reply by nickludlam** [reply]
Thanks. I think I avoided using autosave.strokes.txt because it was not persisted when the stroke count is zero.  That meant if you closed with a clear screen, your previous drawing would always come back. 

https://github.com/pubpub-zz/ppInk/blob/master/src/FormCollection.cs#L4052

In the comment, it's explicit about this behaviour, and I felt it might be more complex to integrate, so I did the much more simple (but bad) separate implementation.  Maybe this could work if it changed to:

```
                if (IC.Ink.Strokes.Count > 0 || Root.Persistent_Drawing)
                    SaveStrokes(st + "AutoSave.strokes.txt");
```

**Reply by pubpub-zz** [reply]
I do not think it is a good idea to add Persistent_Drawing to save the file : with your proposal, if for any reason, you erase the drawings and exit you will loose your work. For me you should just use Persistent_Drawing on restoring data, and perhaps do not restore data if the data (based on file date) is over 1h. If you want to force restoring, you just have to use the restore button. Your opinion ?

**Reply by nickludlam** [reply]
I think you're right, adding `|| Root.Persistent_Drawing` will prevent other behaviour that might be important to some users. For now, I'm happy with the behaviour on my fork, and I have no idea if it's something other people might find useful, so I am hesitant to add a PR.

**Reply by pubpub-zz** [reply]
> I dislike to do some too specific development but as long as you did the job, there is no reason to 'loose it' when I will produce a new version.
Personally I'm so happy to see somebody contributing to an GPL /open source project that I will encourage him😁
